jsonapi_utility.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+)
+
+const (
+	ErrorCodeNotFound        = "not_found_error"
+	ErrorCodeBadParameter    = "bad_parameter_error"
+	ErrorCodeVersionConflict = "version_conflict_error"
+	ErrorCodeUnknownError    = "unknown_error"
+	ErrorCodeConversionError = "conversion_error"
+	ErrorCodeInternalError   = "internal_error"
+)
+
+// ConvertErrorFromModelToJSONAPIError returns the JSONAPI representation
+// of an error from the models package and the HTTP status code that will
+// be associated with it.
+func ConvertErrorFromModelToJSONAPIError(err error) (app.JSONAPIError, int) {
+	detail := err.Error()
+	var title, code string
+	var statusCode int
+	switch err.(type) {
+	case models.NotFoundError:
+		code = ErrorCodeNotFound
+		title = "Not found error"
+		statusCode = http.StatusNotFound
```

Comment:
Well, this is not the actual return code of the HTTP response. In JSONAPI you can have many errors in an error response and each had its own HTTP status code.  See [here](http://jsonapi.org/format/#errors):

> When a server encounters multiple problems for a single request, the most generally applicable HTTP error code SHOULD be used in the response. For instance, 400 Bad Request might be appropriate for multiple 4xx errors or 500 Internal Server Error might be appropriate for multiple 5xx errors.

The `statusCode` here is just the code for the particular error, not for the overall HTTP response.